app/src/main/java/ch/admin/bag/dp3t/storage/DiaryStorage.java

```
@@ -0,0 +1,124 @@
+package ch.admin.bag.dp3t.storage;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import androidx.security.crypto.EncryptedSharedPreferences;
+import androidx.security.crypto.MasterKeys;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+
+import org.crowdnotifier.android.sdk.model.DayDate;
+
+import ch.admin.bag.dp3t.storage.model.DiaryEntry;
+
+public class DiaryStorage {
+
+	private static final String KEY_DIARY_STORE = "KEY_DIARY_STORE";
+	private static final String KEY_DIARY_ENTRIES_V3 = "KEY_DIARY_ENTRIES_V3";
+	private static final Type EXPOSURE_LIST_V3_TYPE = new TypeToken<ArrayList<DiaryEntry>>() { }.getType();
+
+
+	private static DiaryStorage instance;
+
+	private SharedPreferences sharedPreferences;
+	private final Gson gson = new Gson();
+
+	private DiaryStorage(Context context) {
+		try {
+			String KEY_ALIAS = MasterKeys.getOrCreate(MasterKeys.AES256_GCM_SPEC);
+			sharedPreferences = EncryptedSharedPreferences.create(KEY_DIARY_STORE,
+					KEY_ALIAS,
+					context,
+					EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+					EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM);
+		} catch (GeneralSecurityException | IOException ex) {
+			ex.printStackTrace();
```

We should rethrow the exception, not just print it.